LFS171x/docs/introduction-to-hyperledger-composer.md

```diff
@@ -0,0 +1,976 @@
+>
+> Now we install the Node Version Manager or nvm. You may need to reload your bash shell configuration file. Now we can install the LTS version of node.js and activate it.
+>
+> We now download and install the Docker Community Edition. You may need to make an account. We drag Docker into our Applications folder. Now open it and login with your account. Check that it works by running the hello-world image.
+>
+> We now download the VSCode editor. We drag it to our Applications Folder and open it. We go to the Extension section, and search for Hyperledger. We then install the Hyperledger Composer extension.
+>
+> Let us switch back to the terminal. In our bash configuration file we tell NVM to use the LTS version of Node. And we can now use the Node Package Manager to install the Hyperledger Composer tools.
+>
+> After this we download the Hyperledger Fabric development server scripts. We extract the files and we run downloadFabric.sh to fetch the Docker images.
+>
+> We can then start the Hyperledger Fabric development server and create a peerAdmin identity card.
+>
+> Finally, we start the Hyperledger Composer playground, which we will use in a subsequent video.
+
+### Text editor
+On your local machine, we recommend you have an editor featuring a plugin for Hyperledger Composer.
+
+Two such editors exist today:
+
+* [Atom](https://atom.io/)
+* [VS Code](https://code.visualstudio.com)
+
+These will enable you to more easily work using the Hyperledger Composer Extension.
+
+## Installing Hyperledger Composer Components
+
+You will now need to install the Hyperledger Composer components:
+
+This installs the Composer Command Line Interface (CLI)
+
+```
+npm install -g composer-cli
+```
+
+Then install the Composer REST Server
+
+```
+npm install -g composer-rest-server
+```
+
+Install Composer Playground
+```
+npm install -g  composer-playground
+```
+
+Install Yeoman, enabling us to create empty or populated sample blockchain networks and web applications.
+```
+npm install -g yo
+```
+
+Install the Hyperledger Composer generator for Yeoman
+```
+npm install -g generator-hyperledger-composer
+```
+
+## Install Hyperledger Fabric Development Server
+We can now install the Hyperledger Fabric development server, which will act as the backend for our Hyperledger Composer work.
+
+```
+mkdir ~/fabric-tools && cd ~/fabric-tools
+
+curl -O https://raw.githubusercontent.com/hyperledger/composer-tools/master/packages/fabric-dev-servers/fabric-dev-servers.zip
+unzip fabric-dev-servers.zip
+```
+
+We can now download the Docker images for the Hyperledger Fabric components:
+
+```
+./downloadFabric.sh
+```
+
+### Starting Hyperledger Fabric and Composer Playground
+Inside the `fabric-tools` folder, we can now start the Docker images that comprise the Hyperledger Fabric network
+
+```
+./startFabric.sh
+```
```

Comment:
Error here because I had not re-logged in to get docker group. Worth providing a troubleshooting note.
```
./downloadFabric.sh
Development only script for Hyperledger Fabric control
Running 'downloadFabric.sh'
FABRIC_VERSION is unset, assuming hlfv12
FABRIC_START_TIMEOUT is unset, assuming 15 (seconds)
Warning: failed to get default registry endpoint from daemon (Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.38/info: dial unix /var/run/docker.sock: connect: permission denied). Using system default: https://index.docker.io/v1/
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.38/images/create?fromImage=hyperledger%2Ffabric-peer&tag=1.2.0: dial unix /var/run/docker.sock: connect: permission denied
hyperledger@hyperledger-vbox ~/fabric-tools $ groups
hyperledger adm cdrom sudo dip plugdev lpadmin sambashare
```

Reply:
Done.